### PR TITLE
Create MegaRAC-SP-X-BMC.md

### PR DESCRIPTION
**Pull request by WaaromZoMoeilijk**
This is my example config for Megarac IPMI. Great work, I thank everyone for their contributions!

### CODE REVIEW

File: example_configs/MegaRAC-SP-X-BMC.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+2. **General LDAP Settings**:
+    - **Encryption Type**: `SSL` (or No Encryption if preferred)
+    - **Common Name Type**: `FQDN` (or IP if you use a plain IP address to connect to lldap)
+    - **Server Address**: `fqdn.lldap.tld`
+    - **Port**: `6360` (default for SSL, adjust if necessary to default non ssl `3890`)
+
+3. **Authentication** (use read-only bind user):
+    - **Bind DN**: `uid=bind_user,ou=people,dc=example,dc=com`
+    - **Password**: `bind_user-password`
```
Comment (nitnelave) [inline example_configs/MegaRAC-SP-X-BMC.md:26]: Doesn't really matter, but it bothers me that it uses a mix of dash and underscores :D

File: example_configs/MegaRAC-SP-X-BMC.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+4. **Search Configuration**:
+    - **Search Base**: `ou=people,dc=example,dc=com`
+    - **Attribute of User Login**: `uid`
+
+![Screenshot from 2024-03-09 20-11-35](https://github.com/WaaromZoMoeilijk/lldap/assets/13510720/122d5e9f-ed37-4ab0-80e9-a95970127a0c)
```
Comment (nitnelave) [inline example_configs/MegaRAC-SP-X-BMC.md:32]: Can you add it to this repo as well? I think we have an "images" subdirectory for that

File: example_configs/MegaRAC-SP-X-BMC.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+5. **Navigate**: Go to `Settings > External User Settings > LDAP/E-Directory Settings > Role groups`.
+
+6. **Click on empty role group in order to assign a new one**
+
+7. **Role Group - Group Details**:
+    - **Group Name**: `ipmi`
+    - **Group Domain**: `cn=ipmi,ou=groups,dc=example,dc=com`
```
Comment (nitnelave) [inline example_configs/MegaRAC-SP-X-BMC.md:40]: Can you mention that you need to create the group in LLDAP as well?

**Review by nitnelave — CHANGES_REQUESTED** [review]
Can you add a link to this in the readme?

It looks great!

File: example_configs/MegaRAC-SP-X-BMC.md (hunk 3 above)
Comment (WaaromZoMoeilijk) [reply example_configs/MegaRAC-SP-X-BMC.md:40]: Its mentioned under:

### Pre-requisites
- Create and assign the `ipmi` group in LDAP to a (test) user.

File: example_configs/MegaRAC-SP-X-BMC.md (hunk 2 above)
Comment (WaaromZoMoeilijk) [reply example_configs/MegaRAC-SP-X-BMC.md:32]: Yes I've added them in the PR now, hope I've done it the right way, I've changed the links to point to the repo URLs for when the PR is accepted, lmk if this doesn't work.

File: example_configs/MegaRAC-SP-X-BMC.md (hunk 1 above)
Comment (WaaromZoMoeilijk) [reply example_configs/MegaRAC-SP-X-BMC.md:26]: You're right, upon closer inspection my OCD won't allow this neither.

**Comment by WaaromZoMoeilijk** [comment]
Alright, I think we're there. Please let me know if anything needs fixing.

File: example_configs/MegaRAC-SP-X-BMC.md [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+3. **Authentication** (use read-only bind user):
+    - **Bind DN**: `uid=bind_user,ou=people,dc=example,dc=com`
+    - **Password**: `change_bind_user_password`
+
+4. **Search Configuration**:
+    - **Search Base**: `ou=people,dc=example,dc=com`
+    - **Attribute of User Login**: `uid`
+
+![image](https://github.com/lldap/lldap/blob/main/example_configs/images/megarac_user.png)
```
Comment (nitnelave) [inline example_configs/MegaRAC-SP-X-BMC.md:32]: The link should be relative (i.e. just `images/megarac_user.png`). Same for the other one.

**Review by nitnelave — APPROVED** [review]

**Comment by nitnelave** [comment]
One last nit and I'll merge it!

File: example_configs/MegaRAC-SP-X-BMC.md (hunk 4 above)
Comment (WaaromZoMoeilijk) [reply example_configs/MegaRAC-SP-X-BMC.md]: My bad, this explains a lot!
Changed now!

**Comment by WaaromZoMoeilijk** [comment]
Haha, we're getting there, now I believe we're good :)

**Review by nitnelave — APPROVED** [review]

**Comment by nitnelave** [comment]
Thanks!

**Pull request merged**